### ISSUE TITLE
Add methods to fetch the first or last item in a Map

### ISSUE THREAD

**Issue by larry0x** [CosmWasm/cw-storage-plus#35]
I think something like this would be useful:

```rust
// pseudocode
impl<K, V> Map<K, V>
where
    K: PrimaryKey + KeyDeserialize,
    V: Serialize + Deserialize,
{
    /// Return the first item in the map when items are ordered by key in the specified order.
    /// None if the map is empty.
    pub fn first(storage: &dyn Storage, order: Order) -> StdResult<(<K as KeyDeserialize>::Output, V)> {
        // ...
    }

    pub fn last(storage: &dyn Storage, order: Order) -> StdResult<(<K as KeyDeserialize>::Output, V)> {
        // ...
    }
}
```

**Comment by chipshort** [comment]
Would this be different from the following?
```rust
let first = MAP
    .range(&deps.storage, None, None, Order::Ascending)
    .next();

let last = MAP
    .range(&deps.storage, None, None, Order::Descending)
    .next();
```

**Comment by larry0x** [comment]
> Would this be different from the following?
> 
> ```rust
> 
> let first = MAP
> 
>     .range(&deps.storage, None, None, Order::Ascending)
> 
>     .next();
> 
> 
> 
> let last = MAP
> 
>     .range(&deps.storage, None, None, Order::Descending)
> 
>     .next();
> 
> ```

Yeah this is in fact how I would implement it

**Comment by webmaster128** [comment]
Why do you need `order: Order` in this API? Can't `first` just be the first and `last` be the last in ascending order and that's it?

**Comment by larry0x** [comment]
> Why do you need `order: Order` in this API? Can't `first` just be the first and `last` be the last in ascending order and that's it?

True, we don't need `order` here

**Comment by webmaster128** [comment]
Done in https://github.com/CosmWasm/cw-storage-plus/pull/42. Will be released as partof 1.1.
